### PR TITLE
Hexeditor: add option to write hex bytes

### PR DESCRIPTION
**Pull request by AlexThesys**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**
Target widget for this feature: Hexdump. The purpose of this feature is to give user the ability to enter a string of hexadecimal bytes (using Hex Editors RMB context menu) in the form of either 'ff00fccdcd' or '\xff\x00\xfc\xcd\xcd' and that would be exactly what gets written. Although this seems to be quite a common operation, currently there is no way of doing this without using the console. Basically we just parse the string to select the pairs of chars, then convert them to integers with radix 16 and write to the selected location.

**Test plan (required)**
Try entering any number of hexadecimal bytes using RMB->Edit->Write Hex Bytes in Hexdump.
hexedit_demo
The amount of bytes written respects the selected region (if any selection is performed). Any bytes past selection won't be overwritten.
In case, when bytes string is entered in the form of '\x00\x00', this is detected by checking only the first two bytes, so any mistakes in the input won't be corrected.
Incorrect input will result in zeros being written.
If the odd number of nibbles is entered, the last one will be ignored.

### CODE REVIEW

File: src/widgets/HexWidget.cpp [hunk 1]
```diff
@@ -730,6 +734,41 @@ void HexWidget::w_increaseDecrease()
...
+
+    QInputDialog d;
+    d.setInputMode(QInputDialog::InputMode::TextInput);
+    QByteArray bytes = d.getText(this, tr("Write hex bytes"), tr("Hex byte string:"),
+                                 QLineEdit::Normal, "", &ok)
+                               .toUtf8();
+    const int offset = bytes.startsWith("\\x") ? 2 : 0;
+    const int incr = offset + 2;
+    const int bytes_size = qMin(bytes.size() / incr, size);
+    if (ok && bytes_size) {
```
Comment (XVilka) [inline src/widgets/HexWidget.cpp:757]: Probably better to invert the condition and just quickly return if none of these are valid.

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by XVilka — APPROVED** [review]

File: src/widgets/HexWidget.h [hunk 2]
```diff
@@ -315,6 +315,7 @@ private slots:
     // Write command slots
     void w_writeString();
     void w_increaseDecrease();
+	void w_writeBytes();
```
Comment (ITAYC0HEN) [inline src/widgets/HexWidget.h:318]:
```suggestion
    void w_increaseDecrease();
    void w_writeBytes();
```

**Comment by XVilka** [comment]
@AlexThesys could you please rebase the PR on top of the latest `dev`? It should fix problematic CI jobs.

**Comment by AlexThesys** [comment]
Yes, sure. Done.

**Pull request merged**